src/components/Dropdown.tsx

```diff
@@ -0,0 +1,282 @@
+/**
+ * A simple dropdown menu
+ * @author Alessandra De Lucas
+ * @author Gabe Abrams
+ */
+
+// Import React
+import React, { useReducer } from 'react';
+
+// Import FontAwesome
+// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+// import { IconProp } from '@fortawesome/fontawesome-svg-core';
+
+import Variant from '../types/Variant';
+/*------------------------------------------------------------------------*/
+/* -------------------------------- Types ------------------------------- */
+/*------------------------------------------------------------------------*/
+
+// Props definition
+type Props = {
+  // currently selected dropdown item
+  selectedItem: DropdownItem,
```

Comment:
Let's not treat the dropdown like a `select`. Remember that we're creating a generic dropdown here and in IC, we'll be hacking it into a select-style. So please remove this `selectItem`. We'll take this info from the `dropdownButton`